pkg/build/controller/strategy/custom.go

```diff
@@ -22,12 +23,22 @@ type CustomBuildStrategy struct {
 
 // CreateBuildPod creates the pod to be used for the Custom build
 func (bs *CustomBuildStrategy) CreateBuildPod(build *buildapi.Build) (*kapi.Pod, error) {
-	data, err := bs.Codec.Encode(build)
+	strategy := build.Spec.Strategy.CustomStrategy
+
+	codec := bs.Codec
+	if strategy.BuildAPIVersion != "" {
+		interfaces, err := latest.InterfacesFor(strategy.BuildAPIVersion)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get codec for requested API version (%q): %v", strategy.BuildAPIVersion, err)
```

Comment:
Trello'd here: https://trello.com/c/xxdyGxp1/850-provide-a-storage-upgrade-path-for-objects